wordpress-mobile/WordPress-Android
# Use real endpoint for activity type filter

Parent issue #13268

Merge instructions
1. Update FluxC version when https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1807 gets merged
2. Remove "Not ready for merge" label
2. Merge this PR

This PR replace DummyActivityTypeProvider with the actual implementation from FluxC. Corresponding FluxC PR is [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1807). It also starts passing current DateRange filter to the fetchActivityTypes endpoint.

Note:
- It seems the server currently ignores the "after" and "before" dates. We use a different format than Calypso. However, we use this format on all the APIs. I plan to fix this in a separate PR when we decide on the best approach.

To test:
1. Make sure ActivityLogFiltersFeatureConfig is enabled
2. Open Activity Log
3. Click on Activity Type filter
4. Notice, that actual activities get fetched and shown on the UI
5. Notice each row contains number of activity log entries (eg. User (23))
6. Change language of the app to German
7. Open Activity Log
8. Click on Activity Type filter
9. Notice the Activity Types are translated
----------------------------------------
1. Make sure ActivityLogFiltersFeatureConfig is enabled
2. Make sure you are using atomic site or a jetpack site with a paid plan
3. Open Activity Log
4. Click on Activity Type filter
5. Remember what activities and numbers you see
6. Go back and open date range filter
7. Select 17-18th of December
8. Open Activity Type filter and notice the result differs from step 5

PR submission checklist:
